backend/compact-connect/stacks/reporting_stack.py

```diff
@@ -0,0 +1,158 @@
+from __future__ import annotations
+
+from aws_cdk import Duration
+from aws_cdk.aws_cloudwatch import Alarm, ComparisonOperator, Stats, TreatMissingData
+from aws_cdk.aws_cloudwatch_actions import SnsAction
+from aws_cdk.aws_events import Rule, RuleTargetInput, Schedule
+from aws_cdk.aws_events_targets import LambdaFunction
+from aws_cdk.aws_iam import Effect, PolicyStatement
+from aws_cdk.aws_logs import QueryDefinition, QueryString
+from cdk_nag import NagSuppressions
+from common_constructs.nodejs_function import NodejsFunction
+from common_constructs.stack import AppStack
+from constructs import Construct
+
+from stacks import persistent_stack as ps
+
+
+class ReportingStack(AppStack):
+    def __init__(self, scope: Construct, construct_id: str, *, persistent_stack: ps.PersistentStack, **kwargs):
+        super().__init__(scope, construct_id, **kwargs)
+        self._add_ingest_event_reporting_chain(persistent_stack)
+
+    def _add_ingest_event_reporting_chain(self, persistent_stack: ps.PersistentStack):
+        from_address = f'noreply@{persistent_stack.user_email_notifications.email_identity.email_identity_name}'
+
+        # We use a Node.js function in this case because the tool we identified for email report generation,
+        # EmailBuilderJS, is in Node.js. To make utilizing the tool as simple as possible, we opted to not mix
+        # languages in the Lambda.
+        event_collector = NodejsFunction(
+            self,
+            'IngestEventCollector',
+            description='Ingest event collector',
+            lambda_dir='ingest-event-reporter',
+            handler='collectEvents',
+            timeout=Duration.minutes(15),
+            memory_size=2048,
+            environment={
+                'FROM_ADDRESS': from_address,
+                'COMPACT_CONFIGURATION_TABLE_NAME': persistent_stack.compact_configuration_table.table_name,
+                'DATA_EVENT_TABLE_NAME': persistent_stack.data_event_table.table_name,
+                **self.common_env_vars,
+            },
+        )
+        persistent_stack.data_event_table.grant_read_data(event_collector)
+        persistent_stack.compact_configuration_table.grant_read_data(event_collector)
+
+        ses_resources = [
+            persistent_stack.user_email_notifications.email_identity.email_identity_arn,
+            self.format_arn(
+                partition=self.partition,
+                service='ses',
+                region=self.region,
+                account=self.account,
+                resource='configuration-set',
+                resource_name=persistent_stack.user_email_notifications.config_set.configuration_set_name,
+            ),
+        ]
+        # We'll assume that, if it is a sandbox environment, they're in the SES sandbox
```

Comment:
I know I've asked before, but remind me what SES is? Can we spell that out here?